### DISCUSSION TITLE
Expression to select all elements in a group

### DISCUSSION THREAD

**Discussion by james-bungard-bjss** [structurizr/java#350] [Q&A] [answered]
Hi, is there any way to use the `!elements <expression>` DSL construct to select all members of a group?

I tried `element.parent==<group>` and `element.properties[group]==<group>` but both of these result in a `No elements found for expression` error.

**Comment by simonbrowndotje** [comment]
What are you trying to do?

**Reply by james-bungard-bjss** [reply]
Hi Simon, just trying to avoid repetition in the DSL. I have come across several scenarios where all the elements in a group share a common set of relationships, or I want to apply the same tag to all of them. A contrived example:

```
consumer = softwareSystem "Consumer"

externalDataProviders = group "External Data Providers" {
    providerA = softwareSystem "Provider A"
    providerB = softwareSystem "Provider B"
    providerC = softwareSystem "Provider C"
}

!elements element.group==externalDataProviders {
    tags "External" "DataProvider"
    -> consumer "Provides data for"
}
```

Right now I can add a tag manually to each element in the group and use `!elements element.tag==...` to do the rest, but it would be simpler to just filter by group if that were possible.

**Reply by simonbrowndotje** [reply, marked as answer]
Thanks - that's been added; see https://github.com/structurizr/java/issues/351 for details.
